### PR TITLE
feat: [CDS-46434]: Supporting multiline in additional fields of SNOW

### PR DESCRIPTION
**Pull request by boopesh007**
## Describe your changes
image
image
Supporting multiline at additional fields of snow. 

## Checklist
- [x] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/42818)

### CODE REVIEW

**Review by abhinav-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42818)* status: 1 file(s) to review. 1 of 2 approvals obtained.

File: 400-rest/src/test/java/software/wings/sm/states/collaboration/ServiceNowCreateUpdateStateTest.java [hunk 1]
```diff
@@ -218,7 +220,7 @@ private ServiceNowCreateUpdateParams getParamsForAction(ServiceNowAction service
     params.setTicketType(INCIDENT.name());
     params.setUpdateMultiple(true);
     params.setFields(Collections.singletonMap(ServiceNowFields.DESCRIPTION, DESCRIPTION_VALUE));
-    params.setAdditionalFields(Collections.singletonMap("key", "value"));
+    params.setAdditionalFields(new HashMap<>(Map.of("key", "value")));
```
Comment (abhinav-harness) [inline 400-rest/src/test/java/software/wings/sm/states/collaboration/ServiceNowCreateUpdateStateTest.java:223]: why is this change needed?

**Review by lasdox — APPROVED** [review]
> Reviewed 1 of 1 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42818)* status: 1 file(s) to review. 1 of 2 approvals obtained.

File: 400-rest/src/test/java/software/wings/sm/states/collaboration/ServiceNowCreateUpdateStateTest.java (hunk 1 above)
Comment (boopesh007) [reply 400-rest/src/test/java/software/wings/sm/states/collaboration/ServiceNowCreateUpdateStateTest.java:223]: singletonMap is immutable, it was used in the test data, so had to change it to a mutable one.

**Review by vcalasansh — APPROVED** [review]
> Reviewed 1 of 1 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42818)* status: 1. unresolved discussions.

**Review by vcalasansh — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42818)* status: 1. unresolved discussions.

**Comment by Fernando-Dourado** [comment]
trigger ti3

**Review by Fernando-Dourado — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42818#-:-NMFozA0C3_-zbpXV9An:bnfp4nl)* status: 1. unresolved discussions.

**Review by abhinav-harness — COMMENTED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42818)* status: 1. unresolved discussions.

**Review by abhinav-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42818)* status: :shipit: complete! ✓
